### PR TITLE
Poly1305 manual tweaks

### PR DESCRIPTION
**Pull request by mikejsavage**
I didn't make any notes for this one in the big docs feedback issue (must have missed it), so please make sure you agree with my changes

### CODE REVIEW

File: doc/man/man3/crypto_poly1305.3monocypher [hunk 1]
```diff
@@ -144,25 +145,22 @@ The attacker must not be able to guess the key;
...
+Session keys cannot be used for this.
+They are shared and secret, but would be reused when sending multiple
+messages.
+Random numbers cannot be used either as there is no reasonable way to
+share it with the recipient without also revealing it to the attacker.
 .Pp
 The only practical source for the authentication key is a chunk of the
 encryption stream used to encrypt the message.
 That chunk must be
 .Em dedicated
```
Comment (mikejsavage) [inline doc/man/man3/crypto_poly1305.3monocypher]: I don't understand this paragraph.

Does it go like this:

The "encryption stream" is like a random number stream seeded by the encryption key that you XOR with or whatever to encrypt the data. To authenticate a message you pull a 32 byte Poly1305 key from the head of the stream and do not use those bytes to encrypt any data (keyword in the text being "dedicated"), and then you encrypt your message with the rest of the stream

Comment (mikejsavage) [reply doc/man/man3/crypto_poly1305.3monocypher]: I guess that is pretty much what it says in the next paragraph. maybe this one should be taken out

File: doc/man/man3/crypto_poly1305.3monocypher [hunk 2]
```diff
@@ -33,31 +33,30 @@
...
 .Fc
 .Sh DESCRIPTION
-Poly1305 is a one-time message authentication code.
-"One time" means
-the authentication key can be used only once.
-This makes Poly1305
-.Sy easy to mess up .
-On the other hand, Poly1305 is fast, as well as provably secure if
-used correctly.
+Poly1305 is a one-time message authentication code. "One-time" means the
```
Comment (CuleX) [inline doc/man/man3/crypto_poly1305.3monocypher]: Newline after period.

File: doc/man/man3/crypto_poly1305.3monocypher [hunk 3]
```diff
@@ -66,7 +65,6 @@ to compare the received MAC to the output
 .Ss Streaming interface
 .Fn crypto_poly1305_init
 initialises a context.
-The
```
Comment (CuleX) [inline doc/man/man3/crypto_poly1305.3monocypher]: You removed "The" here, but added it above in a similar context. Why?

File: doc/man/man3/crypto_poly1305.3monocypher [hunk 4]
```diff
@@ -125,17 +126,17 @@ crypto_poly1305_final(&ctx, mac);
 These functions implement Poly1305, described in RFC 7539.
 .Sh  SECURITY CONSIDERATIONS
```
Comment (CuleX) [inline doc/man/man3/crypto_poly1305.3monocypher]: A bit late to notice, but double space after macro shouldn't be there.

**Comment by mikejsavage** [comment]
thanks, fixed those

File: doc/man/man3/crypto_poly1305.3monocypher (hunk 1 above)
Comment (LoupVaillant) [reply doc/man/man3/crypto_poly1305.3monocypher]: Yes, that's exactly what it means.

File: doc/man/man3/crypto_poly1305.3monocypher [hunk 5]
```diff
@@ -144,25 +145,22 @@ The attacker must not be able to guess the key;
...
-.Em reused .
-The attacker will recover it and break all security.
-A simple random number cannot be used either: there is no reasonable
-way to give it to the recipient without also revealing it to the
-attacker.
+Session keys cannot be used for this.
+They are shared and secret, but would be reused when sending multiple
+messages.
+Random numbers cannot be used either as there is no reasonable way to
+share it with the recipient without also revealing it to the attacker.
```
Comment (LoupVaillant) [inline doc/man/man3/crypto_poly1305.3monocypher]: > share _them_ with the recipient without also revealing _them_ to the attacker.

I think?

File: doc/man/man3/crypto_poly1305.3monocypher [hunk 6]
```diff
@@ -173,23 +171,18 @@ Take the first 32 bits of that stream as your authentication key, then
...
 .Ss Protection against side channels
 Use
 .Xr crypto_verify16 3monocypher
-To compare message authentication code.
-Avoid standard buffer comparison functions.
-They do not run in constant time, which allows attackers to recover
-the MAC in relatively few tries.
+to compare message authentication codes.
+Avoid standard buffer comparison functions, as they may not run in
+constant time.
```
Comment (LoupVaillant) [inline doc/man/man3/crypto_poly1305.3monocypher]: I agree with the change, but I wonder if we want to add a little additional warning: the problem with standard comparison functions is that timing attacks are not only possible, they are almost *easy*, even over the network. Other algorithms (Argon2d, naive AES, scrypt…) don't run in constant time, yet remain harder to attack.

Maybe add something like "This allows timing attacks" or something? I know it's redundant, and I don't like the ugliness, but I think this is worth emphasising.

File: doc/man/man3/crypto_poly1305.3monocypher (hunk 5 above)
Comment (mikejsavage) [reply doc/man/man3/crypto_poly1305.3monocypher]: yep

File: doc/man/man3/crypto_poly1305.3monocypher (hunk 6 above)
Comment (mikejsavage) [reply doc/man/man3/crypto_poly1305.3monocypher]: yeah I guess if the manual has warnings it should explain them rather than leaving them as mysteries

**Pull request merged**